graph/prim.ts

```diff
@@ -0,0 +1,59 @@
+import { PriorityQueue } from '../data_structures/heap/heap'
+/**
+ * @function prim
+ * @description Compute a minimum spanning tree(MST) of a fully connected weighted undirected graph. The input graph is in adjacency list form. It is a multidimensional array of edges. graph[i] holds the edges for the i'th node. Each edge is a 2-tuple where the 0'th item is the destination node, and the 1'th item is the edge weight.
+ * @Complexity_Analysis
+ * Time complexity: O(Elog(V))
+ * Space Complexity: O(V)
+ * @param {[number, number][][]} graph - The graph in adjacency list form
+ * @return {Edge[], number} - [The edges of the minimum spanning tree, the sum of the weights of the edges in the tree]
+ * @see https://en.wikipedia.org/wiki/Prim%27s_algorithm
+ */
+export const prim = (graph: [number, number][][]): [Edge[], number] => {
+  if (graph.length == 0) {
+    return [[], 0];
+  }
+  let minimum_spanning_tree: Edge[] = [];
+  let total_weight = 0;
+
+  let priorityQueue = new PriorityQueue((e: Edge) => { return e.b }, graph.length, (a: Edge, b: Edge) => { return a.weight < b.weight });
+  let visited = new Set<number>();
+
+  // Start from the 0'th node. For fully connected graphs, we can start from any node and still produce the MST.
+  visited.add(0);
+  add_children(graph, priorityQueue, 0);
+
+  while (!priorityQueue.isEmpty()) {
+    // We always store the previously visited edge in `edge.a`, and the newly visited node in `edge.b`.
```

Comment:
The previously visited node?

Reply:
I edited the comment. Hopefully it makes more sense.